src/analyzer/analyzerbeats.cpp

```diff
@@ -91,24 +102,29 @@ bool AnalyzerBeats::initialize(TrackPointer tio, int sampleRate, int totalSample
     // if we can load a stored track don't reanalyze it
     bool bShouldAnalyze = shouldAnalyze(tio);
 
+
+    DEBUG_ASSERT(!m_pPlugin);
     if (bShouldAnalyze) {
-        if (m_pluginId == mixxx::AnalyzerSoundTouchBeats::pluginInfo().id) {
-            m_pPlugin = std::make_unique<mixxx::AnalyzerSoundTouchBeats>();
-        } else if (m_pluginId == mixxx::AnalyzerQueenMaryBeats::pluginInfo().id) {
+        if (m_pluginId == mixxx::AnalyzerQueenMaryBeats::pluginInfo().id) {
             m_pPlugin = std::make_unique<mixxx::AnalyzerQueenMaryBeats>();
+        } else if (m_pluginId == mixxx::AnalyzerSoundTouchBeats::pluginInfo().id) {
+            m_pPlugin = std::make_unique<mixxx::AnalyzerSoundTouchBeats>();
         } else {
-            m_pPlugin = std::make_unique<mixxx::AnalyzerQueenMaryBeats>();
+            DEBUG_ASSERT(false);
```

Comment:
A debug assertion is not sufficient here, because m_pluginId is read at runtime from a config file. Please use VERIFY_OR_DEBUG_ASSERT and log a warning.

Reply:
The loop at line 74 verifies that the ID is valid or takes the default. 
I will add a comment.